File: SkaiLab.Invoice.Models/InventoryHistory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SkaiLab.Invoice.Models
{
    public class InventoryHistory
    {
        public long LocationId { get; set; }
        public Location Location { get; set; }
        public int Quantity { get; set; }
        public decimal UnitPrice { get; set; }
    }
    public class InventoryHistoryDetail : InventoryHistory
    {
        public DateTime Date { get; set; }
        public string RefNo { get; set; }
        public decimal Amount => Math.Abs(Quantity * UnitPrice);
    }
}

File: SkaiLab.Invoice/ClientApp/src/app/models/inventoryHistory.ts
import {Location} from "./location";

export class InventoryHistory{
    locationId:number;
    location:Location;
    quantity:number;
    unitPrice:number;
    constructor(){
        this.locationId=0;
        this.location=new Location();
        this.quantity=0;
        this.unitPrice=0;
    }
}
export class InventoryHistoryDetail extends InventoryHistory{
    date:Date;
    refNo:string;
    amount:number;
    constructor(){
        super();
        this.date=new Date();
        this.refNo="";
        this.amount=0;
    }
}